simulator/src/main/java/simblock/node/Farmer.java

```diff
@@ -0,0 +1,105 @@
+package simblock.node;
+
+import simblock.block.Block;
+import simblock.block.PoSpaceBlock;
+import simblock.simulator.Simulator;
+import simblock.simulator.Timer;
+import simblock.task.AbstractMintingTask;
+
+import java.math.BigInteger;
+
+import static simblock.settings.SimulationConfiguration.chia_k;
+import static simblock.simulator.Simulator.arriveBlock;
+
+public class Farmer extends Node{
+
+    private int[] PoSpaceCount;
+    private PoSpaceBlock[] Chain;
+    public Farmer(
+            int nodeID, int numConnection, int region, long miningPower, String routingTableName, boolean useCBR, boolean isChurnNode, int depth
+    ) {
+        super(nodeID, numConnection, region, miningPower, routingTableName, useCBR, isChurnNode);
+        PoSpaceCount = new int[depth];
+        Chain = new PoSpaceBlock[chia_k];
+    }
+
+
+    @Override
+    public PoSpaceBlock getCurrentBlock(){
+        return (PoSpaceBlock) super.getCurrentBlock();
+    }
+    @Override
+    public void addToChain(Block newBlock){
+        // TODO may need to have something that removes a VDF task from the simulator
```

Comment:
In the node addToChain method, they remove the minting task from the simulator - I think this is unnecessary for the Chia implementation since the consensus algorithm should throw out any blocks anyways.

Reply:
I think that we shouldn't need to remove anything from the task queue unless we decide to implement the time lords